bindings/experimental/distrdf/test/test_buildranges.py

```diff
@@ -157,7 +157,7 @@ def test_clustered_ranges_with_two_clusters_two_partitions(self):
         different numbers of entries.
 
         """
-        headnode = get_headnode(1)
+        headnode = get_headnode(1, npartitions=None)
```

Comment:
Instead of passing all these `npartitions=None`, why don't we add code in `get_headnode` to check if `npartitions` was passed and, if not, it's set to None.

Reply:
It's a bit subtle because actually `npartitions` will always be passed from the function that gets called when user creates a distributed RDF, namely `make_dataframe` in the Backend (Spark, Dask, etc.).
It's just that in the tests I call directly `get_headnode` which is not supposed to be a user-facing function. So I thought it could be better to have an exception-throwing `kwargs.pop` to be safer.
Nonetheless, it's easy to turn it into a non-exception-throwing `kwargs.pop("npartitions",None)` also in `get_headnode`. What do you think @eguiraud ?

Reply:
IIUC, `npartitions` is not an optional argument -- in real usecases it will always (have to) be passed to `get_headnode`. so the `get_headnode` function is correct in requesting in. all these tests that call `headnode = get_headnode(1, npartitions=None)` maybe just need a fixture?

Reply:
You are right, `npartitions` is not an optional argument in general. In fact, both head node types `TreeHeadNode` and `EmptySourceHeadNode` require it in their `__init__` method with this commit. It is an optional argument for the user, but we still have to pass some value for it internally. I will create a factory function for the test to avoid too many repetitions of the argument

Reply:
If `npartitions` is not optional, note that you can also have a signature like `def get_headnode(npartitions=None, *args, *kwargs)`, you have to make it positional though.

Reply:
it should be just `def get_headnode(npartitions, *args, **kwargs)` in my opinion (`None` is actually not a valid value, only these tests can get away with it)

Reply:
Ok this might be the right signature I was looking for, let me try with Python 2 and 3

Reply:
Latest commit uses the proposed signature. I also like it better this way, thanks for the suggestion!